service/controller/resource/scrapeconfigs/resource.go

```diff
@@ -85,7 +87,23 @@ func getObjectMeta(v interface{}) (metav1.ObjectMeta, error) {
 }
 
 func toSecret(v interface{}, config Config) (*corev1.Secret, error) {
-	scrapeConfigs, err := toData(v, config)
+	cluster, err := key.ToCluster(v)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	var workloadClusterETCDDomain string = ""
+	if "workload_cluster" == key.ClusterType(v) {
+		clusterID := key.ClusterID(cluster)
+		service, err := config.K8sClient.K8sClient().CoreV1().Services(clusterID).Get(context.Background(), "master", metav1.GetOptions{})
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+		if value, ok := service.Annotations["giantswarm.io/etcd-domain"]; ok {
```

Comment:
Just out of curiosity as this is new to me, what is this `master` service and this annotation and where do they come from? I.e. what creates/maintains them?

Reply:
They are created by the provider operators to allow access to the api server from inside the management clusters